### DISCUSSION TITLE
How to count number of entities in collection

### DISCUSSION THREAD

**Discussion by HoangSang1510** [milvus-io/milvus#34362] [Q&A and General discussion] [answered]
Hello, i am using milvus 2.4.0, is there a fast way to get number of entities in my collection or i need to use collection query with output_fields=["count(*)"], thanks you.

**Comment by yhmo** [comment, marked as answer]
In python, the fast way to get number of entities is:
`print(collection.num_entities)`

Every time a segment is persisted, the basic information of the segment is recorded in Etcd, including its row number.
`collection.num_entities` calculates row number from the Etcd, sum up row numbers of all the persisted segments.
But this number doesn't count the deleted items. Let's say a segment has 1000 rows, and you call `collection.delete()` to delete 50 rows from the segment, the `collection.num_entities` always shows 1000 rows for you.
Furthermore, `collection.num_entities` doesn't count non-persisted segments.

`collection.query(output_fields=["count(*)"])` is a query request, executed by query nodes. It counts deleted items, and counts all segments including non-persisted.

**Reply by HoangSang1510** [reply]
Thank you, but I am using MilvusClient and it seems that MilvusClient has no attribute num_entities.

**Reply by yhmo** [reply]
MilvusClient is a new design of pymilvus, some interfaces are not implemented.
